app/models/user.rb

```diff
@@ -43,14 +43,9 @@ def by_name
     end
   end
 
-  def self.find_for_authentication(tainted_conditions)
-    if tainted_conditions[:subdomains].present?
-      local_authority = LocalAuthority.find_by(subdomain: tainted_conditions[:subdomains].first)
-      tainted_conditions.delete(:subdomains)
-      find_first_by_auth_conditions(tainted_conditions.merge(local_authority_id: local_authority.id))
-    else
-      find_first_by_auth_conditions(tainted_conditions)
-    end
+  def self.find_for_database_authentication(tainted_conditions)
+    local_authority_id = LocalAuthority.find_by(subdomain: tainted_conditions[:subdomain]).id
+    find_first_by_auth_conditions(email: tainted_conditions[:email], local_authority_id: local_authority_id)
```

Comment:
The `find_first_by_auth_conditions` method is a class method so you can call it through the `users` association on the `LocalAuthority` model, e.g.

``` ruby
local_authority = LocalAuthority.find_by!(subdomain: tainted_conditions[:subdomain])
local_authority.users.find_first_by_auth_conditions(email: tainted_conditions[:email])
```

Note the use of the `find_by!` method to raise an error if the local authority isn't found.